server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueCounter.java

```diff
@@ -0,0 +1,268 @@
+import static org.sonar.api.measures.CoreMetrics.CRITICAL_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.FALSE_POSITIVE_ISSUES_KEY;
+import static org.sonar.api.measures.CoreMetrics.INFO_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.MAJOR_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.MINOR_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_BLOCKER_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_CRITICAL_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_INFO_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_MAJOR_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_MINOR_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.OPEN_ISSUES_KEY;
+import static org.sonar.api.measures.CoreMetrics.REOPENED_ISSUES_KEY;
+import static org.sonar.api.measures.CoreMetrics.VIOLATIONS_KEY;
+import static org.sonar.api.rule.Severity.BLOCKER;
+import static org.sonar.api.rule.Severity.CRITICAL;
+import static org.sonar.api.rule.Severity.INFO;
+import static org.sonar.api.rule.Severity.MAJOR;
+import static org.sonar.api.rule.Severity.MINOR;
+
+/**
+ * For each component, computes the measures related to number of issues:
+ * <ul>
+ *   <li>unresolved issues</li>
+ *   <li>false-positives</li>
+ *   <li>open issues</li>
+ *   <li>issues per status (open, reopen, confirmed)</li>
+ *   <li>issues per severity (from info to blocker)</li>
+ * </ul>
+ * For each value, the variation on configured periods is also computed.
+ */
+public class IssueCounter extends IssueVisitor {
+
+  private final static Map<String, String> SEVERITY_TO_METRIC_KEY = ImmutableMap.of(
+    BLOCKER, BLOCKER_VIOLATIONS_KEY,
+    CRITICAL, CRITICAL_VIOLATIONS_KEY,
+    MAJOR, MAJOR_VIOLATIONS_KEY,
+    MINOR, MINOR_VIOLATIONS_KEY,
+    INFO, INFO_VIOLATIONS_KEY
+    );
+
+  private final static Map<String, String> SEVERITY_TO_NEW_METRIC_KEY = ImmutableMap.of(
+    BLOCKER, NEW_BLOCKER_VIOLATIONS_KEY,
+    CRITICAL, NEW_CRITICAL_VIOLATIONS_KEY,
+    MAJOR, NEW_MAJOR_VIOLATIONS_KEY,
+    MINOR, NEW_MINOR_VIOLATIONS_KEY,
+    INFO, NEW_INFO_VIOLATIONS_KEY
+    );
+
+  private final PeriodsHolder periodsHolder;
+  private final MetricRepository metricRepository;
+  private final MeasureRepository measureRepository;
+
+  private final Map<Integer, Counters> countersByComponentRef = new HashMap<>();
+  private Counters currentCounters;
+
+  public IssueCounter(PeriodsHolder periodsHolder,
+    MetricRepository metricRepository, MeasureRepository measureRepository) {
+    this.periodsHolder = periodsHolder;
+    this.metricRepository = metricRepository;
+    this.measureRepository = measureRepository;
+  }
+
+  @Override
+  public void beforeComponent(Component component, Tracking tracking) {
+    // TODO optimization no need to instantiate counter if no open issues
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)